server.py

```diff
@@ -1,34 +1,39 @@
+# -*- coding: utf-8 -*-
 import socket
 import sys
 
-ADDR = ('127.0.0.1', 8001) #port 0 may force os to find an open port
+ADDR = ('127.0.0.1', 8000)  # port 0 may force os to find an open port
+
 
 def setup_server():
     server_socket = socket.socket(
-    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP)
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP)
     server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     server_socket.bind(ADDR)
-    server_socket.listen(1) #1 is backlog of connections it will hold onto
+    server_socket.listen(1)
     return server_socket
 
 
 def response_ok():
-    """ returns a response containg an inital response line, header, body"""
-    response_ok = """HTTP/1.1 200 OK\r\n\
-    DATE: Sun, 21 Jul 2001 23:32:15 GTM\r\n\
-    SERVER: Python/2.7.6\r\n\
-    \r\n\
-    """
-    return response_ok.encode('utf-8')
+    """Return a status 200 HTTP response_ok"""
+
+    response_ok = b"""HTTP/1.1 200 OK\r\n
+    DATE: Sun, 21 Jul 2001 23:32:15 GTM\r\n
+    SERVER: Python/2.7.6\r\n
+    \r\n"""
```

Comment:
This is still problematic.  You've got `"\r\n"` at the end of each line, but a multi-line string *also* preserves the *additional* `"\n"` that your editor is hiding from you, so the line terminators here are all wrong.